core/src/main/scala/com/salesforce/op/stages/impl/feature/IDVectorizer.scala

```diff
@@ -0,0 +1,170 @@
+ */
+
+package com.salesforce.op.stages.impl.feature
+
+import com.salesforce.op.{FeatureHistory, UID}
+import com.salesforce.op.features.TransientFeature
+import com.salesforce.op.features.types._
+import com.salesforce.op.stages.base.sequence.{SequenceEstimator, SequenceModel}
+import com.salesforce.op.utils.json.JsonLike
+import com.salesforce.op.utils.spark.{OpVectorColumnMetadata, OpVectorMetadata}
+import com.twitter.algebird.Monoid._
+import com.twitter.algebird.Operators._
+import com.twitter.algebird.Semigroup
+import org.apache.spark.sql.{Dataset, Encoder}
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
+import com.salesforce.op.utils.spark.RichDataset._
+
+import scala.reflect.runtime.universe.TypeTag
+
+/**
+ * Convert a sequence of id features into a vector by detecting categoricals.
+ * A categorical will be represented as a vector consisting of occurrences of top K most common values of that feature
+ * plus occurrences of non top k values and a null indicator (if enabled).
+ * Non-categoricals will be removed.
+ *
+ * @param uid uid for instance
+ */
+class IDVectorizer
+(uid: String = UID[IDVectorizer])
+  extends SequenceEstimator[ID, OPVector](operationName = "idvec", uid = uid)
+    with PivotParams with CleanTextFun with SaveOthersParams
+    with TrackNullsParam with MinSupportParam with MaxCardinalityParams with OneHotFun {
+
+  private implicit val textStatsSeqEnc: Encoder[Array[TextStats]] = ExpressionEncoder[Array[TextStats]]()
+
+
+  def fitFn(dataset: Dataset[Seq[ID#Value]]): SequenceModel[ID, OPVector] = {
+    require(!dataset.isEmpty, "Input dataset cannot be empty")
+
+    val maxCard = $(maxCardinality)
+    val shouldCleanText = $(cleanText)
+
+    implicit val testStatsSG: Semigroup[TextStats] = TextStats.semiGroup(maxCard)
+    val valueStats: Dataset[Array[TextStats]] = dataset.map(_.map(computeTextStats(_, shouldCleanText)).toArray)
+    val aggregatedStats: Array[TextStats] = valueStats.reduce(_ + _)
+
+    val (isCategorical, topValues) = aggregatedStats.map { stats =>
+      val isCategorical = stats.valueCounts.size <= maxCard
+      val topValues = stats.valueCounts
+        .filter { case (_, count) => count >= $(minSupport) }
+        .toSeq.sortBy(v => -v._2 -> v._1)
+        .take($(topK)).map(_._1)
+      isCategorical -> topValues
+    }.unzip
+
+    val idParams = IDVectorizerModelArgs(
+      isCategorical = isCategorical,
+      topValues = topValues,
+      shouldCleanText = shouldCleanText,
+      shouldTrackNulls = $(trackNulls)
+    )
+
+    val vecMetadata = makeVectorMetadata(idParams)
+    setMetadata(vecMetadata.toMetadata)
+
+    new IDVectorizerModel(args = idParams, operationName = operationName, uid = uid)
+
+  }
+
+  private def computeTextStats(text: ID#Value, shouldCleanText: Boolean): TextStats = {
+    val valueCounts = text match {
+      case Some(v) => Map(cleanTextFn(v, shouldCleanText) -> 1)
```

Comment:
I don't think we need to be allowing `shouldCleanText` option at all, since these are ID values.